### ISSUE TITLE
CS not showing found blocks and not paying it to the correct address 

### ISSUE THREAD

**Issue by mario1987** [vagabondan/CoiniumServ#18]
@vagabondan  As I have found a block this is not shared with CS  It is only added to the pool wallet 

The block that I found is updated in the wallet
![naamloos](https://user-images.githubusercontent.com/5938706/29669153-d6171b70-88e2-11e7-85f2-508aeac6783f.jpg)

These are the logs with the time when the block was found 
[BlockFoundProof.TXT](https://github.com/vagabondan/CoiniumServ/files/1248992/BlockFoundProof.TXT)
[CS Debug.TXT](https://github.com/vagabondan/CoiniumServ/files/1248994/CS.Debug.TXT)
[CS packet-20170824.TXT](https://github.com/vagabondan/CoiniumServ/files/1248995/CS.packet-20170824.TXT)
[CS SERVER.TXT](https://github.com/vagabondan/CoiniumServ/files/1248996/CS.SERVER.TXT)

In the debug log of CS it says 

System.NullReferenceException: Object reference not set to an instance of an object.
   at CoiniumServ.Shares.ShareManager.SubmitBlock(IShare share) in C:\Users\Administrator\Desktop\New folder\CoiniumServ-mario-x11-special_edition\src\CoiniumServ\Shares\ShareManager.cs:line 258

I don't know if this causes the problem or not. 

On CS the API is not updated with the information as can be seen in the pic 
![apinoconfirmedblocks](https://user-images.githubusercontent.com/5938706/29669080-9904af90-88e2-11e7-8692-429e829d846c.jpg)
![noblockstatus](https://user-images.githubusercontent.com/5938706/29669183-e93e7d24-88e2-11e7-882d-6baa2a363ac4.jpg)

Also the Last Block  in not updated
![csnolastblock](https://user-images.githubusercontent.com/5938706/29669130-bec51daa-88e2-11e7-906e-856f3657a67d.jpg)

What I think that causes the problem is this config but I am not sure of it
[config in coins folder.txt](https://github.com/vagabondan/CoiniumServ/files/1249006/config.in.coins.folder.txt)

Redis server works fine 
![redis](https://user-images.githubusercontent.com/5938706/29669954-6d2fbfa6-88e5-11e7-86bc-facc2ce6198d.jpg)
![redis-in-action](https://user-images.githubusercontent.com/5938706/29669956-6e75b852-88e5-11e7-88e3-afc30e31522c.jpg)

As for sql  the blocks that were found were not inserted in sql
![sql-foundblocks](https://user-images.githubusercontent.com/5938706/29671082-e22e4e00-88e8-11e7-9c06-91743bc1e488.jpg)
![sql-nopayments](https://user-images.githubusercontent.com/5938706/29671088-e53d399e-88e8-11e7-8fb7-a4dc99ee6773.jpg)
![sql-transactions](https://user-images.githubusercontent.com/5938706/29671094-e79f2c10-88e8-11e7-9bf4-9adab688a5e6.jpg)
![sql-versioninfo](https://user-images.githubusercontent.com/5938706/29671095-e901a1aa-88e8-11e7-83cb-b77ea953e35f.jpg)
![sql-wallet](https://user-images.githubusercontent.com/5938706/29671096-ea0b349e-88e8-11e7-96dc-4e68b306b85d.jpg)
![sql-stats](https://user-images.githubusercontent.com/5938706/29671099-ebfec7de-88e8-11e7-841f-3a0afa7035aa.jpg)

**Comment by mario1987** [comment]
@vagabondan  I just found a block and I think I know what the problem is. CS or wallet the found block to sql.  But I don't know how to solve this. When I look is mysql the account associated for this pool has all privileges on the database.

**Comment by vkynchev** [comment]
@mario1987 Which release are you running? You could try compiling my version, if the problem is gone the fix is somewhere in the commits... I've made 20+ changes so I cannot post PR(but @vagabondan can... if he wants 🙂 ). Also you could start the coin daemon with testnet=1 in the coin config file so you could test block functionality very fast.

**Comment by vkynchev** [comment]
@mario1987 @vagabondan Also there was a problem with `Block` table where `Reward` was not `Decimal`, but `Bool`...

**Comment by mario1987** [comment]
@vkynchev  Does your version do payouts ??  and update the CS about blocks ??  If it does I will give it a try.
But for now I am using unomp I am not relay happy with it but is does work and do payouts.  

I am using the version of vagabondan. tnx for the comment by the way.

**Comment by vkynchev** [comment]
@mario1987 Yes, It does the payouts (LTC & SCORE(scorecoin)) but I have not tested it with RBC(it should work btw)... Currently I am testing with LTC (Scrypt) and after about a month I'll test with DASH (x11)...

This screenshot is from my scorecoin wallet("" -> pool wallet, "user1" -> user which is mining, "vkynchev" -> account for fees). I've setup 0.4% for `"vkynchev"` and 0.1% for `""`(this has to be done to substract payout fees)...
<img width="415" alt="screen shot 2017-08-27 at 12 28 53" src="https://user-images.githubusercontent.com/10143016/29749039-83f800e6-8b23-11e7-8808-42c4021f197b.png">

**Comment by mario1987** [comment]
@vkynchev great I will test it in the following week when i have time. 
Are you running CoiniumServ in docker ??

**Comment by vkynchev** [comment]
@mario1987 Yes, I am running it in docker with docker-compose... Also I've dockerized the coin daemons and the DBs(redis & mysql).

The screenshot is from the scorecoind container 🙂

**Comment by vagabondan** [comment]
@vkynchev Excuse me, I didn't catch [why you are not able to post PR](https://github.com/vagabondan/CoiniumServ/issues/18#issuecomment-325155109)? Develop branch was fully synchronized with yours.

**Comment by vkynchev** [comment]
@vagabondan I made them locally and they were mainly for customization(changing names, info, etc) But currently I have no free time to work on CS...

**Comment by CMvet** [comment]
@vkynchev I'm having the same issue with CS.  I get about 40% orphan blocks on Pokercoin (PoW/PoS), the found blocks only show up in the coin daemon but not in CS.  Mysql and redis all working appropriately and connecting fine to CS.  I'm running Bonesoul's version as I haven't had a chance to compile vagabondans version on windows to see if that one corrects the issue.
